frontend/src/types/tasks.ts

```diff
@@ -0,0 +1,17 @@
+export interface TaskData {
+  id: string;
+  name: string;
+  description: string;
+  index: number;
+}
+
+export interface TaskPostData {
+  name: string;
+  challengeId: number;
+  description: string;
+  index: number;
+}
+
+export interface TaskPutData {
```

Comment:
```suggestion
export interface TaskPutData extends TaskPostData {
```

Since the API request should be the same except for the addition of the `id` field.